EchoPro/survey.py

```diff
@@ -36,741 +23,440 @@ class Survey:
-
-        # add NASC_adult to transect_results_gdf (needs to occur after generate_bin_ds)
-        self.bio_calc.set_adult_NASC()
-
-        # calculate Jolly-Hampton CV
-        self.bio_calc.jollyhampton_cv = self.run_cv_analysis(kriged_data=False)
-
-        # compute abundance and biomass for each length and age bin
-        self.compute_length_age_variables(data="transect")
+        Loads the biological, NASC, and stratification
+        data using parameters obtained from the configuration
+        files. This will generate data attributes associated with the tags
+        defined in both the configuration yml files and the reference CONFIG_MAP
+        and LAYER_NAME_MAP dictionaries.
+        """
+
+        ### Check whether data files defined from the configuration file exists
+        # Generate flat JSON table comprising all configuration parameter names
+        flat_configuration_table = pd.json_normalize(self.config).filter(regex="filename")
+
+        # Parse the flattened configuration table to identify data file names and paths
+        parsed_filenames = flat_configuration_table.values.flatten()
+
+        # Append the root directory to the parsed data file names
+        data_paths = [Path(self.config['data_root_dir'] + '/' + file) for file in parsed_filenames]
+
+        # Evaluate whether either file is missing
+        data_existence = [path.exists() for path in data_paths]
+
+        # Assign the existence status to each configuration file for error evaluation
+        # Error evaluation and print message (if applicable)
+        if not all(data_existence):
+            missing_data = [ files for files, exists in zip( parsed_filenames, data_existence ) if not exists ]
+            raise FileNotFoundError(f"The following data files do not exist: {missing_data}")
+        
+        # Initialize data attributes ! 
+        setattr( self , 'acoustics' , { } )
+        setattr( self , 'biology' , { } )
+        setattr( self , 'spatial' , { } )
+        setattr( self , 'statistics' , { } )
+
+        ### Data validation and import
+        # Iterate through each file defined/detected in 'data_paths' and 'parsed_filenames'
+
+        for file , path in zip(parsed_filenames, data_paths):
+
+            # Parse config to find appropriate settings
+            config_name = flat_configuration_table.apply(lambda row: row[row == file].index, axis=1)
+            
+            # Convert to string
+            config_identifier = config_name[0].tolist()[0].lower()
+
+            # Extract appropriate dataset name from the file path
+            # -- Based on tail-end of file path string to avoid potential 
+            # -- directory name issues
+            dataset = [x for x in [*CONFIG_MAP.keys()] if x.lower() in config_identifier][0]
+            
+            # Determine data layer name
+            datalayer = [x for x in [*CONFIG_MAP.get(dataset)] if x in config_identifier.split('.')][0]
```

Comment:
You can change the loop to be 
```python
for data_type in ["biological", "stratification", "NASC", "kriging"]:
   # proceed to set data_layer
   # -- simpler since you know what it SHOULD be depending on data_type
```
This way the code in L167-179 can be mostly removed.

Reply:
And then I would loop through the regions
```
for region_id in ["US", "CAN"]:
   # Build config_map as in L186-187.
```